src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestDescriptor.cs

```diff
@@ -0,0 +1,142 @@
+        /// <param name="httpRequestData">A structure that represents an outgoing http request.</param>
+        /// <param name="signingCredentials">A security key and algorithm that will be used to sign the (Signed)HttpRequest.</param>
+        public SignedHttpRequestDescriptor(string accessToken, HttpRequestData httpRequestData, SigningCredentials signingCredentials) 
+            : this(accessToken, httpRequestData, signingCredentials, new SignedHttpRequestCreationParameters(), new CallContext())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="SignedHttpRequestDescriptor"/>.
+        /// </summary>
+        /// <remarks>
+        /// <paramref name="accessToken"/> has to contain the 'cnf' claim so that PoP key can be resolved on the validation side.
+        /// https://tools.ietf.org/html/rfc7800#section-3.1
+        /// Default <see cref="SignedHttpRequestCreationParameters"/> and <see cref="CallContext"/> will be created. 
+        /// </remarks>
+        /// <param name="accessToken">An access token that contains the 'cnf' claim.</param>
+        /// <param name="httpRequestData">A structure that represents an outgoing http request.</param>
+        /// <param name="signingCredentials">A security key and algorithm that will be used to sign the (Signed)HttpRequest.</param>
+        /// <param name="signedHttpRequestCreationParameters">A set of parameters required for creating a SignedHttpRequest.</param>
+        public SignedHttpRequestDescriptor(string accessToken, HttpRequestData httpRequestData, SigningCredentials signingCredentials, SignedHttpRequestCreationParameters signedHttpRequestCreationParameters)
+            : this(accessToken, httpRequestData, signingCredentials, signedHttpRequestCreationParameters, new CallContext())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="SignedHttpRequestDescriptor"/>.
+        /// </summary>
+        /// <remarks>
+        /// <paramref name="accessToken"/> has to contain the 'cnf' claim so that PoP key can be resolved on the validation side.
+        /// https://tools.ietf.org/html/rfc7800#section-3.1
+        /// Default <see cref="SignedHttpRequestCreationParameters"/> and <see cref="CallContext"/> will be created. 
+        /// </remarks>
+        /// <param name="accessToken">An access token that contains the 'cnf' claim.</param>
+        /// <param name="httpRequestData">A structure that represents an outgoing http request.</param>
+        /// <param name="signingCredentials">A security key and algorithm that will be used to sign the (Signed)HttpRequest.</param>
+        /// <param name="callContext">An opaque context used to store work when working with authentication artifacts.</param>
+        public SignedHttpRequestDescriptor(string accessToken, HttpRequestData httpRequestData, SigningCredentials signingCredentials, CallContext callContext)
```

Comment:
Context represents a single call.
Since SignedHttpRequestDescriptor could be reused, perhaps it shouldn't be a property, but only passed to SignedHttpRequestHandler methods.

Reply:
It makes sense to decouple CallContext from the descriptor. I'll refactor the code so that the CallContext is passed as a method parameter. If a user didn't create a context, a default one would be created.

Reply:
Addressed with https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1260/commits/a038e36f12bd6e6be52f812f2bbbd2afa3c214ca

Reply:
Addressed with https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1260/commits/a038e36f12bd6e6be52f812f2bbbd2afa3c214ca